### PR TITLE
ADD: GrokLogstashPluginAttributesMapper

### LINKED ISSUE #466: Provide a Mapping from Logstash Configuration files to Data Prepper configurations
**Is your feature request related to a problem? Please describe.**

Beyond just parsing a Logstash configuration grammar, we must also be able map Logstash plugins and settings into Data Prepper plugins and settings.

**Describe the solution you'd like**

Provide a mapping file which can define how a Logstash plugin should map into Data Prepper.

### PR DESCRIPTION
**Pull request by chenqi0805**
Signed-off-by: qchea <qchea@amazon.com>

### Description
This PR adds the custom attribute mapper for grok.
 
### Issues Resolved
#466 
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/GrokLogstashPluginAttributesMapper.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+@SuppressWarnings("unchecked")
```
Comment (dlvenable) [inline data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/GrokLogstashPluginAttributesMapper.java:25]: I generally try to keep these suppressions at tight as possible. I can tell you need it on `mergeMatchAttributes`, and so I'd recommend adding the annotation to that method.

File: data-prepper-logstash-configuration/src/test/java/org/opensearch/dataprepper/logstash/mapping/GrokLogstashPluginAttributesMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        assertThat(actualPluginSettings.get(dataPrepperAttribute), equalTo(value));
+    }
+
+    @Test
+    void mapAttributes_sets_mapped_attributes_merging_multiple_match() {
+        final LogstashAttribute matchMultiKeysLogstashAttribute = prepareHashTypeMatchLogstashAttribute(
+                Arrays.asList(Map.entry("message", "fake message regex 1"), Map.entry("other", "fake other regex")));
+        final LogstashAttribute matchMessageLogstashAttribute2 = prepareArrayTypeMatchLogstashAttribute("message", "fake message regex 2");
+        final List<LogstashAttribute> matchLogstashAttributes = Arrays.asList(matchMultiKeysLogstashAttribute, matchMessageLogstashAttribute2);
+        final Map<String, Object> expectedPluginSettings = new HashMap<String, Object>() {{
```
Comment (dlvenable) [inline data-prepper-logstash-configuration/src/test/java/org/opensearch/dataprepper/logstash/mapping/GrokLogstashPluginAttributesMapperTest.java:70]: Should this be named `expectedMatchSettings`?

**Review by dlvenable — CHANGES_REQUESTED** [review]

**Review by asifsmohammed — APPROVED** [review]
LGTM

**Review by dlvenable — APPROVED** [review]

**Comment by dlvenable** [comment]
Thanks, this looks good!

**Pull request merged**